packages/core/src/external/carequality/ihe-gateway-v2/outbound/xcpd/process/xcpd-response.ts

```diff
@@ -57,30 +73,24 @@ function handlePatientMatchResponse({
 }): OutboundPatientDiscoveryResp {
   const subject1 =
     getPatientRegistryProfile(jsonObj)?.controlActProcess?.subject?.registrationEvent?.subject1;
-  const addr = subject1?.patient?.patientPerson?.addr;
-  const addressLine1 = addr?.streetAddressLine?._text ?? addr?.streetAddressLine;
-  const city = addr?.city?._text ?? addr?.city;
-  const state = addr?.state?._text ?? addr?.state;
-  const postalCode = addr?.postalCode?._text ?? addr?.postalCode;
-  const country = addr?.country?._text ?? addr?.country;
+  const addr = toArray(subject1?.patient?.patientPerson?.addr);
+  const names = toArray(subject1?.patient?.patientPerson?.name);
 
-  const addresses = [
-    {
-      ...(addressLine1 && { line: [addressLine1] }),
-      ...(city && { city }),
-      ...(state && { state }),
-      ...(postalCode && { postalCode: String(postalCode) }),
-      ...(country && { country: country }),
-    },
-  ];
+  const addresses = addr.map((address: Address) => ({
+    line: toArray(address?.streetAddressLine).filter((l): l is string => Boolean(l)),
+    city: address?.city,
+    state: address?.state,
+    postalCode: String(address?.postalCode),
+    country: address?.country,
+  }));
```

Comment:
I made a change or two here, so now if there is no address we will end up with this:

```
"patientResource": {
        "name": [
          {
            "given": [
              "NWHINONE"
            ],
            "family": "NWHINZZZTESTPATIENT"
          }
        ],
        "gender": "male",
        "birthDate": "19810101",
        "address": []
      }
```

Which is correct since address is a required field in the ihe-gateway-sdk schema

Reply:
Nice!

WRT to the first comment (function to convert a single address), I see a 👍 but no code change. Intentional?